runtime/common/src/migrations/local_currency.rs

```diff
@@ -0,0 +1,440 @@
+			} else {
+				log::info!("{LOG_PREFIX} Skipping pool currency migration");
+			}
+
+			weight
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+			assert!(
+				check_local_coupling::<T>(CurrencyOut::get().into(), CurrencyIn::get().into())
+					.unwrap()
+			);
+
+			let pool =
+				pallet_pool_system::Pool::<T>::get(TargetPoolId::get()).expect("Pool should exist");
+			assert!(pool.currency == CurrencyOut::get().into());
+
+			log::info!("{LOG_PREFIX} PRE UPGRADE: Finished");
+
+			Ok(vec![])
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(_: Vec<u8>) -> Result<(), DispatchError> {
+			let pool =
+				pallet_pool_system::Pool::<T>::get(TargetPoolId::get()).expect("Pool should exist");
+			assert!(pool.currency == CurrencyIn::get().into());
+
+			log::info!("{LOG_PREFIX} POST UPGRADE: Finished");
+
+			Ok(())
+		}
+	}
+
+	fn check_local_coupling<T>(from: CurrencyId, to: CurrencyId) -> Option<bool>
+	where
+		T: pallet_pool_system::Config
+			+ orml_asset_registry::Config<CustomMetadata = CustomMetadata, AssetId = CurrencyId>,
+		<T as orml_asset_registry::Config>::AssetId: From<CurrencyId>,
+	{
+		let to_meta = <orml_asset_registry::Pallet<T> as Inspect>::metadata(&to);
+		let from_meta = <orml_asset_registry::Pallet<T> as Inspect>::metadata(&from);
+
+		match (from_meta, to_meta) {
+			(Some(meta), Some(_)) => {
+				let to_currency_id: CurrencyId = meta
+					.additional
+					.local_representation
+					.map(|x| x.try_into().ok())??;
+				if to_currency_id == to {
+					Some(true)
+				} else {
+					log::error!(
+						"{LOG_PREFIX} CurrencyOut does not have CurrencyIn set as local currency"
+					);
+					Some(false)
+				}
+			}
+			(Some(_), None) => {
+				log::error!("{LOG_PREFIX} CurrencyIn is not registered");
+				None
+			}
+			_ => {
+				log::error!("{LOG_PREFIX} CurrencyOut is not registered");
+				None
+			}
+		}
+	}
+}
+
+pub mod add_bidirectional_trading_pair {
+	use cfg_primitives::Balance;
+	#[cfg(feature = "try-runtime")]
+	use cfg_traits::TokenSwaps;
+	use frame_support::dispatch::RawOrigin;
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec::Vec;
+
+	use super::*;
+
+	const LOG_PREFIX: &str = "AddBidirectionalTradingPair";
+
+	pub struct Migration<T, CurrencyOne, CurrencyTwo, MinOrderAmount>(
+		sp_std::marker::PhantomData<(T, CurrencyOne, CurrencyTwo, MinOrderAmount)>,
+	);
+
+	impl<T, CurrencyOne, CurrencyTwo, MinOrderAmount> OnRuntimeUpgrade
+		for Migration<T, CurrencyOne, CurrencyTwo, MinOrderAmount>
+	where
+		T: pallet_order_book::Config<
+			CurrencyId = CurrencyId,
+			BalanceIn = Balance,
+			BalanceOut = Balance,
+		>,
+		CurrencyOne: Get<CurrencyId>,
+		CurrencyTwo: Get<CurrencyId>,
+		MinOrderAmount: Get<Balance>,
+	{
+		fn on_runtime_upgrade() -> Weight {
+			pallet_order_book::Pallet::<T>::add_trading_pair(
+				RawOrigin::Root.into(),
+				CurrencyTwo::get(),
+				CurrencyOne::get(),
+				MinOrderAmount::get(),
```

Comment:
In this case `MinOrderAmount` is denominated in `CurrencyOne`

Reply:
In this particular case, it does not matter as both currencies have the same denomination.

Reply:
That's right 👍🏻